Story/StoryMenuRedux.cs

```diff
@@ -0,0 +1,226 @@
+        internal MenuLabel onlineDifficultyLabel;
+        internal SimplerSymbolButton iHateCheckboxes;
+
+
+        internal SlugcatStats.Name customSelectedSlugcat = SlugcatStats.Name.White;
+
+        public StoryMenuRedux(ProcessManager manager) : base(manager)
+        {
+
+            storyModeOnline = OnlineManager.lobby.gameMode as StoryGameMode;
+
+            StoryMenuHelpers.SanitizeStoryClientSettings(storyModeOnline.storyClientData);
+            StoryMenuHelpers.SanitizeStoryGameMode(storyModeOnline);
+
+            storyModeOnline.currentCampaign = slugcatPages[slugcatPageIndex].slugcatNumber;
+
+            StoryMenuHelpers.RemoveExcessStoryObjects(this, storyModeOnline);
+            StoryMenuHelpers.ModifyExistingMenuItems(this);
+
+            if (OnlineManager.lobby.isOwner)
+            {
+                StoryMenuHelpers.SetupHostMenu(this, storyModeOnline);
+                var hostSettings = StoryMenuHelpers.GetHostBoolStoryRemixSettings();
+                storyModeOnline.storyBoolRemixSettings = hostSettings.hostBoolSettings;
+                storyModeOnline.storyFloatRemixSettings = hostSettings.hostFloatSettings;
+                storyModeOnline.storyIntRemixSettings = hostSettings.hostIntSettings;
+            }
+            else
+            {
+                StoryMenuHelpers.SetupClientMenu(this, storyModeOnline);
+                if (RainMeadow.rainMeadowOptions.SlugcatCustomToggle.Value)
+                {
+                    StoryMenuHelpers.CustomSlugcatSetup(this, storyModeOnline);
+                }
+
+            }
+            SteamSetup();
+            UpdatePlayerList();
+            StoryMenuHelpers.SetupOnlineMenuItems(this, storyModeOnline);
+
+            StoryMenuHelpers.SetupOnlineCustomization(this);
+
+            MatchmakingManager.instance.OnPlayerListReceived += OnlineManager_OnPlayerListReceived;
+
+        }
+
+        public void StartGame()
+        {
+
+            RainMeadow.DebugMe();
+            if (!OnlineManager.lobby.isOwner) // I'm a client
+            {
+                if (ModManager.MMF)
+                {
+                    StoryMenuHelpers.SetClientStoryRemixSettings(storyModeOnline.storyBoolRemixSettings, storyModeOnline.storyFloatRemixSettings, storyModeOnline.storyIntRemixSettings); // Set client remix settings to Host's on StartGame()
+                }
+                if (!RainMeadow.rainMeadowOptions.SlugcatCustomToggle.Value) // I'm a client and I want to match the hosts
+                {
+                    personaSettings.playingAs = storyModeOnline.currentCampaign;
+                }
+            }
+            else //I'm the host
+            {
+
+
+                RainMeadow.Debug("CURRENT CAMPAIGN: " + StoryMenuHelpers.GetCurrentCampaignName(storyModeOnline));
+                personaSettings.playingAs = storyModeOnline.currentCampaign;
+            }
+
+            manager.arenaSitting = null;
+            if (restartChecked)
```

Comment:
pretty sure story menu already had this functionality, can we reuse that?

Reply:
- [ ] Rename Class
- [ ] storyModeOnline => storyGameMode
- [ ] Retrieve built-in sluglist
- [ ] simplify onlineDifficultyLabel
- [ ] Cast client slugcat page depending on save state presence